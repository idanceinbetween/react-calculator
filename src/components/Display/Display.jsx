import React from 'react';
import PropTypes from 'prop-types';
import './Display.css';

const Display = ( {displayValue} ) => {
  return (
      <div className='display-container'>
          {displayValue}
      </div>
  )
}

Display.propTypes = { displayValue: PropTypes.string.isRequired};

export default Display;
