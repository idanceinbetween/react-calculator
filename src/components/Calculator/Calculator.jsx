import React, {Component} from 'react';
import './Calculator.css';
import Display from '../Display/Display';
import Keypad from '../Keypad/Keypad';

class Calculator extends Component {

    state = {
        displayValue: '0',
        numbers: ['9','8','7','6','5','4','3','2','1','0','.','ce'],
        operators: ['/', 'x', '-', '+'],
        selectedOperator: '',
        storedValue: ''
    }

    callOperator = () => console.log('Call Operator')
    setOperator = () => console.log('Set Operator')
    updateDisplay = () => console.log('Update Display')

    render() {
        const { displayValue, numbers, operators} = this.state

        return (
            <div className="calculator-container">
                <Display displayValue={displayValue}/>
                <Keypad
                    callOperator={this.callOperator}
                    numbers={numbers}
                    operators={operators}
                    setOperator={this.setOperator}
                    updateDisplay={this.updateDisplay}
                />
            </div>
        );
    }
}

export default Calculator;