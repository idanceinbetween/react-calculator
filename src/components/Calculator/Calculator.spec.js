import React from 'react';
import { shallow } from 'enzyme';
import Calculator from './Calculator';
import Display from '../Display/Display';
import Keypad from '../Keypad/Keypad';

describe('Calculator', ()=>{
    let wrapper;
    beforeEach(()=> wrapper = shallow(<Calculator />))

    it('renders correctly based on snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    })

    it('renders a div', ()=> {
        expect(wrapper.find('div').length).toEqual(1);
    })

    it('renders the Display and Keypad components', () => {
        expect(wrapper.containsAllMatchingElements([<Display displayValue={wrapper.instance().state.displayValue}/>, <Keypad
            callOperator={wrapper.instance().callOperator}
            numbers={wrapper.instance().state.numbers}
            operators={wrapper.instance().state.operators}
            setOperator={wrapper.instance().setOperator}
            updateDisplay={wrapper.instance().updateDisplay}
        />])).toEqual(true);
    })
})