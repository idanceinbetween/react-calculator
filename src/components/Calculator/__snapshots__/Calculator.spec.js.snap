// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator renders correctly based on snapshot 1`] = `ShallowWrapper {}`;
